benchmarks/engines/busyring/arbor/ring.cpp

```diff
@@ -336,6 +325,82 @@ double interp(const std::array<T,2>& r, unsigned i, unsigned n) {
 }
 
 arb::cable_cell branch_cell(arb::cell_gid_type gid, const cell_parameters& params) {
+    arb::sample_tree tree;
+
+    // Add soma.
+    double soma_radius = 12.6157/2.0;
+    tree.append(arb::mnpos, {{0,0,0,soma_radius}, 1}); // For area of 500 μm².
+
+    std::vector<std::vector<unsigned>> levels;
+    levels.push_back({0});
+
+    // Standard mersenne_twister_engine seeded with gid.
+    std::mt19937 gen(gid);
+    std::uniform_real_distribution<double> dis(0, 1);
+
+    double dend_radius = 0.5; // Diameter of 1 μm for each cable.
+
+    double dist_from_soma = soma_radius;
+    for (unsigned i=0; i<params.max_depth; ++i) {
+        // Branch prob at this level.
+        double bp = interp(params.branch_probs, i, params.max_depth);
+        // Length at this level.
+        double l = interp(params.lengths, i, params.max_depth);
+        // Number of compartments at this level.
+        unsigned nc = std::round(interp(params.compartments, i, params.max_depth));
+
+        std::vector<unsigned> sec_ids;
+        for (unsigned sec: levels[i]) {
+            for (unsigned j=0; j<2; ++j) {
+                if (dis(gen)<bp) {
+                    auto z = dist_from_soma;
+                    auto p = tree.append(sec, {{0,0,z,dend_radius}, 3});
+                    if (nc>1) {
+                        auto dz = l/nc;
+                        for (unsigned k=1; k<nc; ++k) {
+                            p = tree.append(p, {{0,0,z+k*dz, dend_radius}, 3});
+                        }
+                    }
+                    sec_ids.push_back(tree.append(p, {{0,0,z+l,dend_radius}, 3}));
+                }
+            }
+        }
+        if (sec_ids.empty()) {
+            break;
+        }
+        levels.push_back(sec_ids);
+
+        dist_from_soma += l;
+    }
+
+    arb::label_dict d;
+
+    using arb::reg::tagged;
+    d.set("soma",      tagged(1));
+    d.set("dendrites", join(tagged(3), tagged(4)));
```

Comment:
Is there anything with tag 4?

Reply:
No, there isn't. This code was cut and paste from elsewhere, which used the SWC convention of dendrites being 3 and 4.